### PR TITLE
Add nf-test to CI

### LINKED ISSUE #22: Run nf-test in CI
### Description of feature

At the moment, the CI only runs the pipeline using test data, but not nf-test. It would be more adequate to run nf-test automatically on every PR.

### PR DESCRIPTION
**Pull request by kedhammar**
## PR checklist

- [x] This comment contains a description of changes (with reason).
  - Add nf-test to CI, based on [the nf-test docs](https://www.nf-test.com/tutorials/github-actions/?h=github)
  - Closes #22 
- ~[ ] If you've fixed a bug or added code that should be tested, add tests!~
- ~[ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/seqinspector/tree/master/.github/CONTRIBUTING.md)~
- ~[ ] If necessary, also make a PR on the nf-core/seqinspector _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.~
- [x] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- ~[ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).~
- ~[ ] Usage Documentation in `docs/usage.md` is updated.~
- ~[ ] Output Documentation in `docs/output.md` is updated.~
- [x] `CHANGELOG.md` is updated.
- ~[] README.md` is updated (including new tool citations and authors/contributors).~

### CODE REVIEW

**Comment by kedhammar** [comment]
Since I've opted for conservative nf-test runs I'm running it manually to make sure we have full coverage.

 I’m running into a single, very simple assertion error for the utils_nextflow_pipeline subworkflow.
 
```
  Test [dd2e962c] 'Should print version' Assertion failed: 

assert workflow.stdout.contains("nextflow_workflow v9.9.9")
       |        |      |
       workflow []     false

FAILED (8.614s)

  Assertion failed: 
  
  1 of 2 assertions failed
  
  Nextflow stdout:
  
  Nextflow stderr:
```
  
This is after running nf-core submodules update as well. I've asked about it in the `nf-test` slack channel now but it might not be our problem.

**Comment by kedhammar** [comment]
Ping @maxulysse

**Comment by kedhammar** [comment]
Got confirmation from @mirpedrol it's an issue on the `nf-core/modules` repo

**Comment by Aratz** [comment]
Awesome! I've tried to make some changes on my own repo to test it but I get this weird error in the ci :thinking: Is this something you've encountered yourself?
https://github.com/Aratz/seqinspector/actions/runs/11573234518/job/32214895326

File: .github/workflows/nf-test.yml
Comment (maxulysse) [inline .github/workflows/nf-test.yml:1]: why not in the ci.yml file?

Comment (kedhammar) [reply .github/workflows/nf-test.yml:1]: The ci.yml file runs the whole pipeline if I understand correctly.

I'd prefer to have them as separate GHA checks, if they do different things.

**Review by Aratz — APPROVED** [review]
Great!

Just one thing is that now it's running the tests for the modules we're using, even though these are supposed to be validated in https://github.com/nf-core/modules/. I think for now it's fine but it might get very slow as we add more and more modules. Maybe we can look into how to restrict nf-test. I tried to look it up now but didn't manage to go anywhere...

**Pull request merged**